applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -145,28 +144,57 @@ static void handle_dvfs_error(int32_t err)
 	(void) k_work_reschedule(&dvfs_retry.retry_work, timeout);
 }
 
-static void set_dvfs_freq(enum dvfs_frequency_setting target_freq)
+static void dvfs_notify_cb(void)
+{
+	int res;
+	int32_t ret = sys_notify_fetch_result(&cli.notify, &res);
+
+	if (ret < 0) {
+		LOG_DBG("work not completed");
+		return;
+	}
+
+	ret = nrf_clock_control_release(dvfs_clock_dev, &spec);
+	if (ret < 0) {
+		LOG_ERR("Failed to release requested clock specs");
+		module_set_state(MODULE_STATE_ERROR);
+		module_state = STATE_ERROR;
+		cancel_dvfs_retry_work();
+		return;
+	}
+
+	sys_notify_init_callback(&cli.notify, dvfs_notify_cb);
```

Comment:
Is it one shot?
If it is used for only the requests, why not register it only right before the nrf_clock_control_request instead of here and in case of the MODULE_STATE_READY?

Reply:
Yes it is one shot. We need to register it again on reuse (before making the subsequent request).

Reply:
yes it is one shot.

Reply:
> why not register it only right before the nrf_clock_control_request instead of here and in case of the MODULE_STATE_READY?

what about it?